src/model/chunk.py

```diff
@@ -0,0 +1,42 @@
+## This file contains the function to chunk the input text into smaller pieces for the model to process
+## The input text is tokenized and then split into chunks of size chunk_size
+
+from transformers import AutoTokenizer
+import torch
+
+def chunk(inp: str, 
+          tokenizer: AutoTokenizer, 
+          chunk_size: int = 256, 
+          overlapping_len: int = 3,
+          max_chunks: int = 128) -> torch.Tensor:
+    
+    # Tokenize entire sample
+    tokenized_txt = tokenizer(inp,
+                              return_tensors="pt"
+                             )["input_ids"].view(-1)
+    token_len = len(tokenized_txt)
+    
+    # Add chunks
+    chunks = []
+
+    for i in range(0, token_len-overlapping_len, chunk_size-overlapping_len):
+        # Exit if max_chunks is met
+        if len(chunks) >= max_chunks:
+            break
+
+        # Create (potentially too short) new chunk
+        new_chunk = tokenized_txt[i:i+chunk_size]
+        
+        # Generate (potentially empty) padding
+        padding = torch.full(
+            size=(chunk_size - len(new_chunk), ),
+            fill_value=tokenizer.pad_token_id
```

Comment:
If `tokenizer.pad_token_id` is None, this breaks. The previously mentioned check or at least an assertion at the top of this function is necessary

Code as simple as the one below breaks this code
```py
from chunk import chunk
from transformers import AutoTokenizer
from datasets import load_dataset

ds = load_dataset("codeparrot/github-code", streaming=True, split="train", trust_remote_code=True)

tokenizer = AutoTokenizer.from_pretrained("gpt2")

sample = next(iter(ds))["code"]

print(chunk(sample, tokenizer))```
```
Simple fix

```suggestion
            fill_value=tokenizer.pad_token_id if tokenizer.pad_token_id else tokenizer.eos_token_id